model/Validator/SyncParamsValidator.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA ;
+ */
+
+namespace oat\taoSync\model\Validator;
+
+use oat\oatbox\service\ConfigurableService;
+use oat\taoSync\model\VirtualMachine\SupportedVmService;
+use oat\taoSync\model\Exception\SyncRequestFailedException;
+
+class SyncParamsValidator extends ConfigurableService
+{
+    const SERVICE_ID = 'taoSync/SyncParamsValidator';
+
+    /**
+     * @param array $parameters
+     * @throws SyncRequestFailedException
+     */
+    public function validate(array $parameters)
+    {
+        if (!isset($parameters['tao_version'])) {
+            throw new SyncRequestFailedException('Required parameter "tao_version" is missing.');
+        }
+
+        $supportedVmService = $this->getServiceLocator()->get(SupportedVmService::SERVICE_ID);
+        $supportedVmVersions = $supportedVmService->getSupportedVmVersions();
+
+        if (!in_array($parameters['tao_version'], $supportedVmVersions)) {
```

Comment:
In the frontend the form field is called "Minimum version", which implies that client VM's with greater version are allowed, while here the check is for exact version. I guess either the label should be changed or functionality updated.